app/models/name/format.rb

```diff
@@ -96,10 +96,9 @@ def self.group_suffix(name)
   # Make sure display names are in boldface for accepted names, and not in
   # boldface for deprecated names.
   def self.make_sure_names_are_bolded_correctly
-    Name.connection.select_values(%(
-      SELECT id FROM names
-      WHERE IF(deprecated, display_name LIKE "%*%", display_name NOT LIKE "%*%")
-    )).map do |id|
+    select_manager = arel_select_accepted_and_deprecated
+    # puts(select_manager.to_sql)
+    Name.connection.select_values(select_manager.to_sql).map do |id|
```

Comment:
I can't improve upon Arel in this case, but since we're going to load each Name instance anyway, why not use find_by_sql?

Actually, I remember you didn't like using named_function (for the IF in this case).  How about doing it as two queries instead:

``
(Name.where(deprecated: true, Name[:display_name].matches("%*%")).to_a + 
 Name.where(deprecated:false).and.not(Name[:display_name].matches("%*%"))).to_a).each do |name|
  name.changed_deprecated(name.deprecated)
  name.save
end
``

Using matches just to check if there is an asterisk in a string always seemed overkill.  There is a function CHARINDEX... would that be better to use?  Are there any db-agnostic equivalents?

Reply:
I'll stick with `matches` for now. 

AR/Arel mixing notes: We apparently can't stack AR next to Arel inside the `where` - it has to be two separate `where`s. Also, it didn't like joining the arrays for some reason, so I just made it an `or`. This also reduces it back to a single query, instead of the two.

```ruby
Name.where(deprecated: true).where(Name[:display_name.matches("%*%")).
or(Name.where(deprecated: false).where(Name[:display_name].does_not_match("%*%")))
```